tests/unit/test_runtime.py

```diff
@@ -0,0 +1,184 @@
+"""Test the EventStreamRuntime, which connects to the RuntimeClient running in the sandbox."""
+
+import os
+import pathlib
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from opendevin.core.config import SandboxConfig
+from opendevin.events import EventStream
+from opendevin.events.action import (
+    CmdRunAction,
+)
+from opendevin.events.observation import (
+    CmdOutputObservation,
+)
+from opendevin.runtime.client.runtime import EventStreamRuntime
+from opendevin.runtime.plugins import AgentSkillsRequirement, JupyterRequirement
+from opendevin.runtime.server.runtime import ServerRuntime
+
+
+@pytest.fixture
+def temp_dir(monkeypatch):
+    # get a temporary directory
+    with tempfile.TemporaryDirectory() as temp_dir:
+        pathlib.Path().mkdir(parents=True, exist_ok=True)
+        yield temp_dir
+
+
+async def _load_runtime(box_class, event_stream, plugins, sid):
+    sandbox_config = SandboxConfig()
+    if box_class == EventStreamRuntime:
+        runtime = EventStreamRuntime(
+            sandbox_config=sandbox_config,
+            event_stream=event_stream,
+            sid=sid,
+            # NOTE: we probably don't have a default container image `/sandbox` for the event stream runtime
+            # Instead, we will pre-build a suite of container images with OD-runtime-cli installed.
+            container_image='ubuntu:22.04',
```

Comment:
I am little confused about this point. Does this mean we need to prebuild everything when launching? I though we do some work on sandbox agnostic in previous PR to build image?

Reply:
When `DEBUG=false` (by default), we only need to pre-build once (e.g., copying all the OD code into the container and `poetry install` all the dependencies).

However, when we are developing, especially if you change the code (e.g., `client.py`) that is supposed to be run *inside* the container, you would need a way to *update* the code inside the container and install any missing dependencies. This is supported by setting `DEBUG=true` - every time you run the code, you will update the existing image with the latest code.

There will be three layers of images:
- Original Image: `ubuntu:22.04`
- Runtime Image (build from stable release of OpenDevin): `od_runtime:ubuntu__22.04` - this is the runtime image that the user will use
- Dev Runtime Image (build from local source): `od_runtime_dev:ubuntu__22.04`

When `DEBUG=true,` Dev Runtime Image will be re-build every time from `od_runtime:ubuntu__22.04`: most dependencies are already installed; we just need to update the code and dependency (if any). It will take ~10 seconds, so it is usually fast.

Reply:
Yeah, 10 seconds is fast. I know the od_runtime and od_runtime_dev in previous PR. Get your point now.